configs/components/gem-prune.rb

```diff
@@ -4,6 +4,6 @@
   pkg.add_source('file://resources/rubygems-prune')
 
   pkg.build do
-    "GEM_PATH=\"#{settings[:host_gem]}\" RUBYOPT=\"-Irubygems-prune\" #{settings[:host_gem]} prune"
+    "GEM_PATH=\"#{settings[:gem_home]}\" RUBYOPT=\"-Irubygems-prune\" #{settings[:host_gem]} prune"
```

Comment:
Where is this set?

Reply:
It's a good question, it appears RubyGems sets it to a ruby version in the lib directory in the project (and there's only one Ruby version for bolt). It comes out to `/opt/puppetlabs/bolt/lib/ruby/gems/2.7.0` in my run of vanagon build, here's an output of all the settings:

```
{:puppet_runtime_version=>"202205060", :puppet_runtime_location=>"https://builds.delivery.puppetlabs.net/puppet-runtime/202205060/artifacts/", :puppet_runtime_basename=>"bolt-runtime-202205060.el-7-x86_64", :runtime_project=>"bolt", :ruby_version=>"2.7.6", :openssl_version=>"1.1.1", :rubygem_net_ssh_version=>"6.1.0", :augeas_version=>"1.11.0", :rubygem_gettext_version=>"3.2.9", :rubygem_deep_merge_version=>"1.2.1", :rubygem_puppet_version=>"7.13.1", :prefix=>"/opt/puppetlabs/bolt", :ruby_dir=>"/opt/puppetlabs/bolt", :bindir=>"/opt/puppetlabs/bolt/bin", :ruby_bindir=>"/opt/puppetlabs/bolt/bin", :libdir=>"/opt/puppetlabs/bolt/lib", :includedir=>"/opt/puppetlabs/bolt/include", :datadir=>"/opt/puppetlabs/bolt/share", :mandir=>"/opt/puppetlabs/bolt/share/man", :host_ruby=>"/opt/puppetlabs/bolt/bin/ruby", :host_gem=>"/opt/puppetlabs/bolt/bin/gem", :gem_home=>"/opt/puppetlabs/bolt/lib/ruby/gems/2.7.0", :gem_install=>"/opt/puppetlabs/bolt/bin/gem install --no-document --local --bindir=/opt/puppetlabs/bolt/bin", :platform_triple=>nil, :host=>nil, :artifactory_url=>"https://artifactory.delivery.puppetlabs.net/artifactory", :buildsources_url=>"https://artifactory.delivery.puppetlabs.net/artifactory/generic/buildsources", :cppflags=>"-I/opt/puppetlabs/bolt/include -I/opt/pl-build-tools/include", :cflags=>"-I/opt/puppetlabs/bolt/include -I/opt/pl-build-tools/include", :ldflags=>"-L/opt/puppetlabs/bolt/lib -L/opt/pl-build-tools/lib -Wl,-rpath=/opt/puppetlabs/bolt/lib", :openssl_extra_configure_flags=>["no-dtls", "no-dtls1", "no-idea", "no-seed", "no-weak-ssl-ciphers", "-DOPENSSL_NO_HEARTBEATS"], :verbose=>false, :skipcheck=>false, :link_bindir=>"/opt/puppetlabs/bin", :main_bin=>"/usr/local/bin"}
```

I was working off of pdk-vanagon's use of `:host_gem` https://github.com/puppetlabs/pdk-vanagon/commit/3b3eb60eb40459e27d4309f71f6623c99b070eef#diff-0edf00f9a9a053cc0ba7507fe0479067d8405638a7187316f86405b82c860266R13 which also relies on RubyGems's setting, but we could make this more explicitly set.

As a sidenote, I am mystified by some settings, do you know where 'settings[:privatedir]' comes from in pdk-vanagon? It does not appear to be set nor a RubyGems thing.